### PR TITLE
13205 - Fix for BC Online account creation, don't allow empty BCOL details. A…

### PR DESCRIPTION
**Pull request by seeker25**
…lso correctly create an account if LINKED.

Also fix it so account details don't bleed over from parent account.

*Issue #:*
https://github.com/bcgov/entity/issues/13205
https://github.com/bcgov/entity/issues/13970

*Description of changes:*
A bit of this is fairly rough - I think it should be rewritten when we move to Vue 3. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/create-account/PaymentMethodSelector.vue [hunk 1]
```diff
@@ -101,14 +121,39 @@ export default class PaymentMethodSelector extends Mixins(Steppable) {
     this.isPADValid = isValid
   }
 
-  private save () {
+  private async save () {
     this.setCurrentOrganizationPaymentType(this.selectedPaymentMethod)
-    this.createAccount()
+    try {
```
Comment (seeker25) [inline auth-web/src/components/auth/create-account/PaymentMethodSelector.vue:127]: Similar functionality exists in the code

File: auth-web/src/components/auth/create-account/AccountTypeSelector.vue [hunk 2]
```diff
@@ -176,7 +176,7 @@ export default class AccountTypeSelector extends Mixins(Steppable) {
 
   private async mounted () {
     // first time to the page , start afresh..this is Create New account flow
-    if (!this.currentOrganization) {
+    if (this.currentOrganization) {
```
Comment (seeker25) [inline auth-web/src/components/auth/create-account/AccountTypeSelector.vue:179]: Account details bleed over fix.

File: auth-web/src/components/auth/create-account/AccountCreatePremium.vue [hunk 3]
```diff
@@ -242,6 +243,7 @@ export default class AccountCreatePremium extends Mixins(Steppable) {
       bcolAccountName: details.bcolAccountDetails.orgName
     }
     this.setCurrentOrganization(org)
+    this.setCurrentOrganizationPaymentType(PaymentTypes.BCOL)
```
Comment (seeker25) [inline auth-web/src/components/auth/create-account/AccountCreatePremium.vue:246]: Makes sense to set payment to BCOL, considering the user just typed in their USER / pass to fetch their previous account details.

File: auth-web/src/components/auth/create-account/PaymentMethodSelector.vue [hunk 4]
```diff
@@ -88,8 +103,13 @@ export default class PaymentMethodSelector extends Mixins(Steppable) {
   }
 
   private get isEnableCreateBtn () {
-    return (this.selectedPaymentMethod === PaymentTypes.PAD)
-      ? (this.selectedPaymentMethod && this.isPADValid) : !!this.selectedPaymentMethod
+    if (this.selectedPaymentMethod === PaymentTypes.PAD) {
+      return this.isPADValid
+    } else if (this.selectedPaymentMethod === PaymentTypes.BCOL) {
+      return !(this.currentOrganization.bcolProfile?.password === undefined || this.currentOrganization.bcolProfile?.password === '')
```
Comment (seeker25) [inline auth-web/src/components/auth/create-account/PaymentMethodSelector.vue:109]: I'm not a huge fan of using emits, I think we should be using state instead throughout the code

File: auth-web/src/store/modules/org.ts [hunk 5]
```diff
@@ -314,6 +314,11 @@ export default class OrgModule extends VuexModule {
     this.currentOrganization = { ...this.currentOrganization, ...orgBusinessType }
   }
 
+  @Mutation
+  public setCurrentOrganizationBcolProfile (bcolProfile: BcolProfile) {
+    this.currentOrganization = { ...this.currentOrganization, bcolProfile }
```
Comment (kialj876) [inline auth-web/src/store/modules/org.ts:319]: should this be `...bcolProfile` ?

File: auth-web/src/components/auth/create-account/PaymentMethodSelector.vue [hunk 6]
```diff
@@ -101,14 +121,40 @@ export default class PaymentMethodSelector extends Mixins(Steppable) {
...
+      this.createAccount()
+    }
   }
 
   @Emit('final-step-action')
   private createAccount () {
   }
+
+  @Emit('emit-bcol-info')
+  private getBcolInfo (bcolProfile: BcolProfile) {
```
Comment (kialj876) [inline auth-web/src/components/auth/create-account/PaymentMethodSelector.vue:155]: does it make more sense to call this `setBcolInfo` ?

File: auth-web/src/components/auth/create-account/PaymentMethodSelector.vue (hunk 4 above)
Comment (kialj876) [inline auth-web/src/components/auth/create-account/PaymentMethodSelector.vue:109]: would `!!this.currentOrganization.bcolProfile?.password` work instead?

**Review by kialj876 — COMMENTED** [review]
lgtm just a couple minor things to consider

File: auth-web/src/store/modules/org.ts (hunk 5 above)
Comment (seeker25) [reply auth-web/src/store/modules/org.ts:319]: No, ...bcolProfile would spread `password` and `username` to the organization - `organization.password` `organization.username` which wouldn't be right

File: auth-web/src/components/auth/create-account/PaymentMethodSelector.vue (hunk 6 above)
Comment (seeker25) [reply auth-web/src/components/auth/create-account/PaymentMethodSelector.vue:155]: I think so, ideally I'd like to get rid of these emits

File: auth-web/src/components/auth/create-account/PaymentMethodSelector.vue (hunk 4 above)
Comment (seeker25) [reply auth-web/src/components/auth/create-account/PaymentMethodSelector.vue:109]: Yes

**Review by kialj876 — APPROVED** [review]

**Comment by seeker25** [comment]
There's some issues with this ticket, I'll save it for next release. Trying to get a release out today.

**Pull request merged**